File: src/main/java/mil/nga/tiff/compression/RawCompression.java
package mil.nga.tiff.compression;

import java.nio.ByteOrder;

/**
 * Raw / no compression
 *
 * @author osbornb
 */
public class RawCompression implements CompressionDecoder, CompressionEncoder {

    @Override
    public byte[] decode(byte[] bytes, ByteOrder byteOrder) {
        return bytes;
    }

    @Override
    public boolean rowEncoding() {
        return false;
    }

    @Override
    public byte[] encode(byte[] bytes, ByteOrder byteOrder) {
        return bytes;
    }

}

File: src/main/java/mil/nga/tiff/compression/RawCompression.kt
package mil.nga.tiff.compression

import java.nio.ByteOrder

/**
 * Raw / no compression
 */
class RawCompression : CompressionDecoder, CompressionEncoder {
    override fun decode(bytes: ByteArray, byteOrder: ByteOrder): ByteArray {
        return bytes
    }

    override fun rowEncoding(): Boolean {
        return false
    }

    override fun encode(bytes: ByteArray, byteOrder: ByteOrder): ByteArray {
        return bytes
    }
}
